crates/bevy_render/src/pipeline/vertex_buffer_descriptor.rs

```diff
@@ -1,37 +1,36 @@
 use super::VertexFormat;
-use bevy_utils::HashMap;
-use std::borrow::Cow;
-
-pub use bevy_derive::AsVertexBufferDescriptor;
+use std::{
+    borrow::Cow,
+    hash::{Hash, Hasher},
+};
 
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct VertexBufferDescriptor {
     pub name: Cow<'static, str>,
     pub stride: u64,
     pub step_mode: InputStepMode,
-    pub attributes: Vec<VertexAttributeDescriptor>,
+    pub attribute: VertexAttributeDescriptor,
```

Comment:
I doubt this is a suprise, but I would like to point out that this is now a "higher level" abstraction that makes us diverge from wgpu / how these apis actually work under the hood. In general I would prefer it if we didn't make these apis opinionated, as it prevents us from building alternative abstractions later (ex: the interleaved approach we will almost certainly implement in the future).

Is there a way to leave this as a `Vec<VertexAttributeDescriptor>`, then make our "opinionated assertions" at a higher level? I'd rather panic or return an error when `attributes.len() != 1` in places where we need to make that assumption.